relayer/src/link/relay_path.rs

```diff
@@ -1438,62 +1598,90 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
             (true_res, false_res)
         }
 
-        let connection_delay = self.channel.connection_delay;
+        let src_chain_time = self.src_time_latest()?;
+        let src_max_block_time = self.src_max_block_time()?;
+        let src_latest_height = self.src_latest_height()?;
```

Comment:
I am a bit concerned that we add the overhead of getting this data from chain for 0-delay paths which are the overwhelming majority of cases...and these calls are not needed there. Could we skip these for the `self.channel.connection_delay.is_zero()` case:
```suggestion
        let (elapsed_src_ods, unelapsed_src_ods) = if self.channel.connection_delay.is_zero() {
            (self.src_operational_data.take(), VecDeque::new())
        } else {
            let src_chain_time = self.src_time_latest()?;
            let src_max_block_time = self.src_max_block_time()?;
            let src_latest_height = self.src_latest_height()?;
            ...
```

Reply:
Thanks for noticing this! I took a slightly different approach from the one you suggested here. I used closures to evaluate the queries lazily and only if required. There are two cases where we don't need to handle the connection delay - one when it is zero and one where it is non-zero, but the `OperationalData` doesn't have packet messages. So IMHO the closure approach is better because it has the added benefit that it handles all these cases with the same code, but please let me know if you think otherwise and I will change it to handle these cases separately.  https://github.com/informalsystems/ibc-rs/pull/1917/files#diff-2a5a700c0b1975ad9d64941997aa53e0b6e26e577fa8f1e9a2aba4abe7cf1f95R1604

Reply:
Yes, that works better!